### DISCUSSION TITLE
offtopic: implicit weights in DUCET. can I ask for some advice?

### DISCUSSION THREAD

**Discussion by gpawru** [unicode-org/icu4x#4623] [General]
Hello! I'm really, REALLY sorry for going off-topic. I understand that this might not be entirely relevant, but perhaps you could help me figure it out?

I'm writing a collator, and for the Tangut language, the weights are implicit.

In this case, I'm considering the DUCET, without delving into the CLDR territory.

In the specification ([TR#10](https://www.unicode.org/reports/tr10/tr10-49.html#Implicit_Weights)), it's stated:

Siniform ideographic scripts: Tangut = Assigned code points in Block=`Tangut` OR Block=`Tangut_Components` OR Block=`Tangut_Supplement`

So, as I understand it, it includes `U+17000` ..= `U+18AFF` and `Tangut_Supplement`.

If we trust the UCD [Blocks.txt](https://unicode.org/Public/UCD/latest/ucd/Blocks.txt), `Tangut_Supplement` ends at `U+18D7F`. If we trust UCA DUCET [allkeys.txt](https://unicode.org/Public/UCA/15.1.0/allkeys.txt), `Tangut_Supplement` ends at `U+18D8F`.

So is it `U+18D7F` or `U+18D8F`?
Or should one calculate implicit weights according to the scheme for `Tangut_Supplement` for `U+18D00` ..= `U+18D08`, and consider `U+18D09` as Unassigned (which seems logical, but the confusion about UCD/UCA remains)?

I understand that the issue is minor, and perhaps not an issue at all, but if there's a mistake somewhere here, it would be good to correct it. If not, and it's just me being a bit stupid, I would really appreciate comments from smart and knowledgeable people in this matter.

Thank you!

**Comment by Manishearth** [comment]
I believe 7F is correct. Not sure what's going on with DUCET.

But since the area in question is unassigned anyway, I don't think it should matter for your use case.

**Reply by gpawru** [reply]
Sometimes I myself start to think that my meticulousness is excessive. But in fact, the inaccuracies made now can lead to problems that will arise later :) Thank you for your response, have a good day!

**Reply by Manishearth** [reply]
Got a proper answer, Unicode 14.0 decided to size down the block, and it wasn't propagated to all files. Hopefully will be fixed in the Unicode 16 files.

**Reply by gpawru** [reply]
Huge thank you!
